### ISSUE TITLE
Penetration volume

### ISSUE THREAD

**Issue by zskuang58** [jyunlee/InterHandGen#5]
It seems that there is an error in the calculation of the penetration volume metric.  When directly inferring with the official pre-trained model you provided, the penetration volume metric yields 0. Could you please offer some guidance or suggestions on this matter?

**Comment by On-JungWoan** [comment]
If there are no penetration cases, the loss_col value yields 0. Adding this code might help to solve that issue.

```python
            if loss_col == 0:
                grad_col += torch.zeros_like(x0_t_)
```

**Comment by jyunlee** [comment]
Hello, thank you for your interest in our work! :)

Could you try removing `try` block in line 174 of `evaluate.py` and check if you have any unexpected error for penetration computation (e.g., `trimesh` library not installed)? When I recently computed the penetration volume using the pre-trained model, I obtained 0.75 using a random seed "19960911".

**Comment by zskuang58** [comment]
Thanks for the author's reply, I have tried it, but it seems that it is still calculated as 0.

**Comment by Erwin2233** [comment]
I have same issues

**Comment by Erwin2233** [comment]
> Hello, thank you for your interest in our work! :)
> 
> Could you try removing `try` block in line 174 of `evaluate.py` and check if you have any unexpected error for penetration computation (e.g., `trimesh` library not installed)? When I recently computed the penetration volume using the pre-trained model, I obtained 0.75 using a random seed "19960911".

    raise ValueError("Not all meshes are volumes!")
